cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -353,6 +359,30 @@ public void drop(Id.Stream streamId) throws Exception {
   }
 
   @Override
+  public boolean createOrUpdateView(Id.Stream.View viewId, ViewSpecification spec) throws Exception {
+    Preconditions.checkArgument(exists(viewId.getStream()), "Stream '%s' does not exist.", viewId.getStreamId());
```

Comment:
All these actions should be done through the `StreamCoordinatorClient`, which provides locking mechanism for stream admin operations. You might want to add a new method to `StreamCoordinatorClient` so that it can perform arbitrary operation while holding the lock. Something like:

```java
interface StreamCoordinatorClient {

  <T> T exclusiveAction(Id.Stream streamId, Callable<T> action) throws Exception;

  void exclusiveAction(Id.Stream streamId, Runnable action) throws Exception;
}
```